src/lava/lib/optimization/solvers/generic/read_gate/process.py

```diff
@@ -45,6 +45,8 @@ def __init__(self,
                          name=name,
                          log_config=log_config)
         self.target_cost = Var(shape=(1,), init=target_cost)
+        self.best_solution = Var(shape=shape, init=-1)
+        self.best_solution_step = Var(shape=(1,), init=-1)

```

Comment:
Not sure if these need to be actual Vars (or LavaTypes in the ProcModel). IMHO, we can just define them as nparrays in the PyProcModel and not mention them in the Process. Vars are only necessary if you want to somehow interact with them from externally, like by defining their initial value or reading them during runtime.

Or am I missing something, @srrisbud ?

Reply:
These are used in the CProcModel to store the best solution (and associated time step) across different iterations. Am I missing something or is it right to add them to the Process?